services/api-server/src/simcore_service_api_server/api/routes/files.py

```diff
@@ -60,6 +67,40 @@
 }
 
 
+async def _get_file(
+    *,
+    file_id: UUID,
+    storage_client: StorageApi,
+    user_id: int,
+    access_right: AccessRight,
+):
+    """Gets metadata for a given file resource"""
+
+    try:
+        stored_files: list[StorageFileMetaData] = await storage_client.search_files(
+            user_id=user_id,
+            file_id=file_id,
+            sha256_checksum=None,
+            access_right=access_right,
+        )
+        if not stored_files:
+            msg = "Not found in storage"
+            raise ValueError(msg)  # noqa: TRY301
+
+        stored_file_meta = stored_files[0]
+        assert stored_file_meta.file_id  # nosec
```

Comment:
The search function can return multiple entries, but in this case it shouldn't, because the UUID should characterize a single element. However, I only factored out this function in order to be able to reuse it. Hence, for now I prefer to keep it as it was in order to not break anything.

Reply:
OK with not changing the function.
All I'm saying is to put an extra assert before line 90. This will be stripped in prod code, and will run with tests.
This should not change the behaviour and will give us more guarantees for the future.

Reply:
OK, I see your point. I added the assert. Thanks for pointing it out.